design/ceph/multus-network.md

```diff
@@ -78,21 +78,71 @@ Nothing to do in particular since they don't use any service IPs.
-
-Restarting the node will cause the multus interface in the host namespace to go away. The holder pod will once again have the interface, which will be remigrated via the setup job.
-
-When a new node is added, a holder pod is added to it on behalf of the daemonset, and the migration process above occurs. 
-
-The holder pod, the setup job, and the teardown job require a service account with the permissions to interact with pods and jobs. The rook-csi-multus service account was created to fulfill this need.
-
-The holder daemonset is deleted when the CephCluster is marked for deletion. The termination of the holder pods trigger the teardown jobs to run, removing the multus interfaces from the host network namespaces of the nodes. 
+Steps must be taken to fix a CSI-with-multus issue documented
+[here](https://github.com/rook/rook/issues/8085). To summarize the issue:
+when a CephFS/RBD volume is mounted in a Pod using cephcsi and then the CSI CephFS/RBD plugin is
+restarted or terminated (e.g. by restarting or deleting its DaemonSet), all operations on the volume
+become blocked, even after restarting the CSI pods. The only workaround is to restart the node where
+the cephcsi plugin pod was restarted.
+
+When deploying a CephCluster resource configured to use multus networks, a multus-connected network
+interface will be added to the host network namespace for all nodes that will run CSI plugin pods.
+
+The Rook-Ceph Operator's CSI controller creates a DaemonSet configured to use the
+`network.selectors.public` network specified for the CephCluster. This DaemonSet runs on all the
+nodes that will have CSI pods. Its Pods exist to "hold" a particular network interface that the CSI
+pods can reliably connect to. These Pods will be referred to as the "holder" Pods.
+
+TODO: VERIFY THIS, MAYBE IT'S UNECCESSARY: These Pods should only be stopped and restarted when a
+node is stopped so that volume operations do not become blocked. The Rook-Ceph Operator's CSI
+controller should place a finalizer on the Pods so that they cannot be accidentally removed, and we
+should make note that even if the holder DaemonSet is updated, the Pods will only be changed on a
+node reboot.
+
+Furthermore, a new Rook-specific CSI sidecar will run in CSI plugin Pods alongside the Ceph CSI
+plugin process. This sidecar will watch for holder Pods, and when a holder Pod is created or
+updated, it will move the public network from the Pod's network namespace to the host node's network
+namespace. In this way, even if the CSI pods are restarted, the network interface will not change,
+and the operations to the Ceph volume will not be blocked.
+
+If a holder Pod is removed, the sidecar should remove the interface from the host's network
+namespace. Similarly, if the sidecar container is stopped, it should move all holder Pod namespaces
```

Comment:
Why move everything back? What if the pod is restarted. Moving the interface back to the holder pod will likely break client connections to Ceph.

Reply:
I should investigate how pod restarts work. If the pod is deleted (during update, for example) I think we lose the "hold" on the interface. But if it's just restarted, I wonder if we would get any events...